Modules/Microsoft365DSC/DSCResources/MSFT_PPTenantSettings/MSFT_PPTenantSettings.psm1

```diff
@@ -0,0 +1,490 @@
+      GlobalAdminAccount                             = $GlobalAdminAccount
+    }
+  }
+  catch
+  {
+    try
+    {
+      Write-Verbose -Message $_
+      $tenantIdValue = ''
+      if (-not [System.String]::IsNullOrEmpty($TenantId))
+      {
+        $tenantIdValue = $TenantId
+      }
+      elseif ($null -ne $GlobalAdminAccount)
+      {
+        $tenantIdValue = $GlobalAdminAccount.UserName.Split('@')[1]
+      }
+      Add-M365DSCEvent -Message $_ -EntryType 'Error' `
+        -EventID 1 -Source $($MyInvocation.MyCommand.Source) `
+        -TenantId $tenantIdValue
+    }
+    catch
+    {
+      Write-Verbose -Message $_
+    }
+    throw $_
+  }
+}
+
+function Set-TargetResource
+{
+  [CmdletBinding()]
+  param (
+    [Parameter(Mandatory = $true)]
+    [ValidateSet('Yes')]
+    [System.String]
+    $IsSingleInstance,
+
+    [Parameter()]
+    [System.Boolean]
+    $WalkMeOptOut,
+
+    [Parameter()]
+    [System.Boolean]
+    $DisableNPSCommentsReachout,
+
+    [Parameter()]
+    [System.Boolean]
+    $DisableNewsletterSendout,
+
+    [Parameter()]
+    [System.Boolean]
+    $DisableEnvironmentCreationByNonAdminUsers,
+
+    [Parameter()]
+    [System.Boolean]
+    $DisablePortalsCreationByNonAdminUsers,
+
+    [Parameter()]
+    [System.Boolean]
+    $DisableSurveyFeedback,
+
+    [Parameter()]
+    [System.Boolean]
+    $DisableTrialEnvironmentCreationByNonAdminUsers,
+
+    [Parameter()]
+    [System.Boolean]
+    $DisableCapacityAllocationByEnvironmentAdmins,
+
+    [Parameter()]
+    [System.Boolean]
+    $DisableSupportTicketsVisibleByAllUsers,
+
+    [Parameter()]
+    [System.Boolean]
+    $DisableDocsSearch,
+
+    [Parameter()]
+    [System.Boolean]
+    $DisableCommunitySearch,
+
+    [Parameter()]
+    [System.Boolean]
+    $DisableBingVideoSearch,
+
+    [Parameter()]
+    [System.Boolean]
+    $DisableShareWithEveryone,
+
+    [Parameter()]
+    [System.Boolean]
+    $EnableGuestsToMake,
+
+    [Parameter()]
+    [ValidateRange(1, 10000)]
+    [System.UInt32]
+    $ShareWithColleaguesUserLimit,
+
+    [Parameter()]
+    [System.Management.Automation.PSCredential]
+    $GlobalAdminAccount
+  )
+
+  Write-Verbose -Message 'Setting Power Platform Tenant Settings configuration'
+
+  #region Telemetry
+  $ResourceName = $MyInvocation.MyCommand.ModuleName.Replace('MSFT_', '')
+  $data = [System.Collections.Generic.Dictionary[[String], [String]]]::new()
+  $data.Add('Resource', $ResourceName)
+  $data.Add('Method', $MyInvocation.MyCommand)
+  $data.Add('Principal', $GlobalAdminAccount.UserName)
+  $data.Add('TenantId', $TenantId)
+  Add-M365DSCTelemetryEvent -Data $data
+  #endregion
+
+  $ConnectionMode = New-M365DSCConnection -Workload 'PowerPlatforms' `
+    -InboundParameters $PSBoundParameters
+
+  $SetParameters = $PSBoundParameters
+  $RequestBody = Get-M365DSCPowerPlatformTenantSettingsJSON -Parameters $SetParameters
+
+  Set-TenantSettings -RequestBody $RequestBody
```

Comment:
Please pipe this out to 
| Out-Null

To prevent unwanted verbose output

Reply:
Fixed in new commit